scripts/zones/Horlais_Peak/mobs/Evil_Oscar.lua

```diff
@@ -51,16 +62,27 @@ evilOscarFillsHisLungs = function(mob)
     if someoneIsAlive then
         local ebbBreathCounter = mob:getLocalVar("EBB_BREATH_COUNTER")
         if ebbBreathCounter < 2 then -- Charge two breaths
-            sendMessage(players)
+            sendFillingMessage(players)
             mob:setLocalVar("EBB_BREATH_COUNTER", ebbBreathCounter + 1)
-        else -- On the third breath, fire straight away!
-            sendMessage(players)
+            mob:timer(math.random(10000, 20000), evilOscarFillsHisLungs)
+        else -- On the third breath use EBB
+            sendFillingMessage(players)
             mob:setLocalVar("EBB_BREATH_COUNTER", 0)
-            mob:useMobAbility(1332)
-        end
+            -- 2 second delay between final message and EBB
+            mob:timer(1500, function(mobForBreath)
+                if mobForBreath:isAlive() then
+                    if mobForBreath:checkDistance(mobForBreath:getTarget()) <= 10.0 then
+                        mobForBreath:useMobAbility(1332)
+                    else
+                        -- use a normal out of range message for now
+                        sendOutOfRangeMessage(mobForBreath:getBattlefield():getPlayers(), mobForBreath:getTarget())
+                    end
```

Comment:
These if else statements should be handled in the ability's lua, no?

Reply:
> These if else statements should be handled in the ability's lua, no?

They are specific to just Evil Oscar (as other mobs that use EBB do not have such a mechanism), the reason for such check is that we want to print an out of range message even if player is out of range when calling useMobAbility which would normally just not print anything.